sdk/search/azure-search/src/main/java/com/azure/search/implementation/SynonymMapsImpl.java

```diff
@@ -60,28 +63,33 @@ public SynonymMapsImpl(SearchServiceRestClientImpl client) {
      * SearchServiceRestClientSynonymMaps to be used by the proxy service to
      * perform REST calls.
      */
-    @Host("https://{searchServiceName}.{searchDnsSuffix}")
+    @Host("{endpoint}")
     @ServiceInterface(name = "SearchServiceRestClientSynonymMaps")
     private interface SynonymMapsService {
         @Put("synonymmaps('{synonymMapName}')")
         @ExpectedResponses({200, 201})
-        Mono<SimpleResponse<SynonymMap>> createOrUpdate(@PathParam("synonymMapName") String synonymMapName, @HostParam("searchServiceName") String searchServiceName, @HostParam("searchDnsSuffix") String searchDnsSuffix, @BodyParam("application/json; charset=utf-8") SynonymMap synonymMap, @HeaderParam("Prefer") String prefer, @QueryParam("api-version") String apiVersion, @HeaderParam("client-request-id") UUID clientRequestId, @HeaderParam("If-Match") String ifMatch, @HeaderParam("If-None-Match") String ifNoneMatch, Context context);
+        @UnexpectedResponseExceptionType(SearchErrorException.class)
+        Mono<SimpleResponse<SynonymMap>> createOrUpdate(@PathParam("synonymMapName") String synonymMapName, @HostParam("endpoint") String endpoint, @BodyParam("application/json; charset=utf-8") SynonymMap synonymMap, @HeaderParam("Prefer") String prefer, @QueryParam("api-version") String apiVersion, @HeaderParam("x-ms-client-request-id") UUID xMsClientRequestId, @HeaderParam("If-Match") String ifMatch, @HeaderParam("If-None-Match") String ifNoneMatch, Context context);
 
         @Delete("synonymmaps('{synonymMapName}')")
         @ExpectedResponses({204, 404})
-        Mono<SimpleResponse<Void>> delete(@PathParam("synonymMapName") String synonymMapName, @HostParam("searchServiceName") String searchServiceName, @HostParam("searchDnsSuffix") String searchDnsSuffix, @QueryParam("api-version") String apiVersion, @HeaderParam("client-request-id") UUID clientRequestId, @HeaderParam("If-Match") String ifMatch, @HeaderParam("If-None-Match") String ifNoneMatch, Context context);
+        @UnexpectedResponseExceptionType(SearchErrorException.class)
+        Mono<Response<Void>> delete(@PathParam("synonymMapName") String synonymMapName, @HostParam("endpoint") String endpoint, @QueryParam("api-version") String apiVersion, @HeaderParam("x-ms-client-request-id") UUID xMsClientRequestId, @HeaderParam("If-Match") String ifMatch, @HeaderParam("If-None-Match") String ifNoneMatch, Context context);
```

Comment:
Any reason that just delete changed from `SimpleResponse<Void>` to `Response<Void>`?

Reply:
It is newly introduced feature in azure core. My understanding here we do not need to check Value in simpleResponse. So Response<Void> is enough and simple for use.